yaml/_loader/loader.ts

```diff
@@ -330,10 +284,7 @@ function storeMappingPair(
         return throwError(state, "nested arrays are not supported inside keys");
       }
 
-      if (
-        typeof keyNode === "object" &&
-        _class(keyNode[index]) === "[object Object]"
-      ) {
+      if (typeof keyNode === "object" && common.isObject(keyNode[index])) {
```

Comment:
This part is also safe to keep as `Object.prototype.toString.call(keyNode[index]) === "[object Object]"`. `common.isObject` is true for arrays, functions, etc, but it's not with the existing code.